core/deployment/src/main/java/io/quarkus/deployment/ApplicationConfig.java

```diff
@@ -7,6 +7,13 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_TIME)
 public class ApplicationConfig {
 
+    /**
+     * The group of the application.
+     * If not set, defaults to the group of the project.
+     */
+    @ConfigItem
+    public String group;
```

Comment:
Here again we're mixing up concepts - why do we need a group concept for this PR?

Reply:
Because it affects the generation of the kubernetes resources and the creation of the docker image in exactly the same way as `name` does.
It feels a little from a user perspective to use `quarkus.application.name` and then `quarkus.kubernetes.group`. 
If you want me to change it, I will obviously do so :)

Reply:
Most docker tags consist of two naming components:

1. group, user etc
2. name

For plain local docker builds we can omit the first (the group bit). But if we are to integrate with docker registries etc, we will have to use one (or we won't be able to perform push). So, I'd say its better to have it there up front than try to incorporate it later.

Note: This group is not related to the `groupId`.

Reply:
Thanks for pointing out how ap4k uses the default group.
Removing the group from the PR is the easy part 😎